### PR TITLE
Support PR trigger of ARM JIT stress jobs

### PR DESCRIPTION
**Pull request by BruceForstall**
Add a list of jobs known to the ARM64 CI, and allow those to be triggered.

Also, (1) allow Release triggered jobs to run tests (before, the label said
"Build and Test", but no test was done), (2) let arm64 debug do a PRI 1 test
build, like ARM32 (reduce differences between these similar tasks).

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Comment by BruceForstall** [comment]
@jashook @RussKeldorph PTAL

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

File: netci.groovy [hunk 1]
```diff
@@ -1133,30 +1178,28 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
...
                                 Utilities.addDefaultPrivateGithubPRTriggerForBranch(job, branch, contextString, null, arm64Users)
                             }
                             break
-                        case 'pri1r2r':
-                        case 'gcstress0x3':
-                        case 'gcstress0xc':
-                            // Stress jobs will will run on private trigger and run tests.
-                            Utilities.addPrivateGithubPRTriggerForBranch(job, branch, contextString,
-                            "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}\\W+${scenario}.*", null, arm64Users)
+                        default:
```
Comment (jashook) [inline netci.groovy]: Nit: add a comment mentioning stress jobs will use this code path.

File: netci.groovy [hunk 2]
```diff
@@ -1242,11 +1285,11 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
                                 Utilities.addDefaultPrivateGithubPRTriggerForBranch(job, branch, contextString, null, arm64Users)
                             }
                             break
-                        case 'pri1r2r':
-                        case 'gcstress0x3':
-                        case 'gcstress0xc':
-                            Utilities.addPrivateGithubPRTriggerForBranch(job, branch, contextString,
-                            "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}\\W+${scenario}.*", null, arm64Users)
+                        default:
```
Comment (jashook) [inline netci.groovy]: Same nit as the arm32 change

File: netci.groovy [hunk 3]
```diff
@@ -1693,25 +1720,24 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
...
-                    assert (scenario == 'default') || (scenario == 'pri1r2r') || (scenario == 'gcstress0x3') || (scenario == 'gcstress0xc')
+                    assert Constants.validArmWindowsScenarios.contains(scenario)
                    
                     // Set time out
                     setTestJobTimeOut(newJob, scenario)
 
                     // Debug runs take too long to run. So build job only.
                     if (lowerConfiguration == "debug") {
-                       buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${architecture} toolset_dir C:\\ats2"
+                       buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${architecture} toolset_dir C:\\ats2 -priority=1"
```
Comment (jashook) [inline netci.groovy]: This seems unnecessary. Wont we be adding a bunch of time to the jump building tests we will never run?

Comment (BruceForstall) [reply netci.groovy]: I can revert this if you like. It just seemed weird to me that we build the pri 1 tests for checked, and for debug for arm, but not here. Why not be consistent?

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Comment by BruceForstall** [comment]
@RussKeldorph @jashook PTAL -- I've changed it so all the new stress periodic jobs for arm/armlb/arm64 are `@weekly`, not `@daily`. Also addressed PR feedback.

File: netci.groovy (hunk 3 above)
Comment (jashook) [reply netci.groovy]: I think I would prefer to keep this as it is, I understand the consistency argument. However, we do not use these tests and the time added to the build is significant.

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Pull request merged**